src/components/Drawer/index.js

```diff
@@ -138,6 +138,7 @@ export default function Drawer(props) {
                     <Header id={headerId} content={header} />
                     <RenderIf isTrue={!hideCloseButton}>
                         <StyledCloseButton
+                            data-id="drawer-close-button"
```

Comment:
since this is a ButtonIcon data-id will not work because it does not have a data-id prop, instead you should use id prop